### ISSUE TITLE
Update of mitemp2_bt fails with timeout

### ISSUE THREAD

**Issue by jPPuetz** [gallynero/mitemp2_bt#1]
Hello,

my system settings are has described here: https://github.com/home-assistant/home-assistant/issues/16456#issuecomment-578487934

I got the following issue with the new HA custom component. 

configuration.yaml:
`sensor:
  - platform: mitemp2_bt
    mac: 'A4:C1:38:47:A4:5E'
    name: Sleeping Room
    timeout: 60
    median: 1
    monitored_conditions:
      - temperature
      - humidity
      - battery
`

Logs:
`
2020-02-27 08:30:33 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for mitemp2_bt which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you do experience issues with Home Assistant.
2020-02-27 08:31:17 WARNING (MainThread) [homeassistant.helpers.entity] Update of sensor.sleeping_room_humidity is taking over 10 seconds
2020-02-27 08:31:38 WARNING (MainThread) [homeassistant.components.sensor] Updating mitemp2_bt sensor took longer than the scheduled update interval 0:00:30
2020-02-27 08:32:09 WARNING (MainThread) [homeassistant.components.sensor] Updating mitemp2_bt sensor took longer than the scheduled update interval 0:00:30
2020-02-27 08:32:09 WARNING (SyncWorker_3) [custom_components.mitemp2_bt.sensor] Polling error Could not read data from Mi Temp sensor A4:C1:38:***
2020-02-27 08:32:19 WARNING (MainThread) [homeassistant.helpers.entity] Update of sensor.sleeping_room_temperature is taking over 10 seconds
2020-02-27 08:32:40 WARNING (MainThread) [homeassistant.components.sensor] Updating mitemp2_bt sensor took longer than the scheduled update interval 0:00:30
2020-02-27 08:33:09 WARNING (SyncWorker_4) [custom_components.mitemp2_bt.sensor] Polling error Could not read data from Mi Temp sensor A4:C1:38:***
2020-02-27 08:33:21 WARNING (MainThread) [homeassistant.helpers.entity] Update of sensor.sleeping_room_temperature is taking over 10 seconds
2020-02-27 08:33:42 WARNING (MainThread) [homeassistant.components.sensor] Updating mitemp2_bt sensor took longer than the scheduled update interval 0:00:30
2020-02-27 08:34:11 WARNING (SyncWorker_19) [custom_components.mitemp2_bt.sensor] Polling error Could not read data from Mi Temp sensor A4:C1:38:***
2020-02-27 08:34:13 WARNING (MainThread) [homeassistant.components.sensor] Updating mitemp2_bt sensor took longer than the scheduled update interval 0:00:30
2020-02-27 08:34:21 WARNING (MainThread) [homeassistant.helpers.entity] Update of sensor.sleeping_room_humidity is taking over 10 seconds
2020-02-27 08:34:44 WARNING (MainThread) [homeassistant.components.sensor] Updating mitemp2_bt sensor took longer than the scheduled update interval 0:00:30
2020-02-27 08:35:11 WARNING (SyncWorker_2) [custom_components.mitemp2_bt.sensor] Polling error Could not read data from Mi Temp sensor A4:C1:38:***
2020-02-27 08:35:25 WARNING (MainThread) [homeassistant.helpers.entity] Update of sensor.sleeping_room_humidity is taking over 10 seconds
2020-02-27 08:35:46 WARNING (MainThread) [homeassistant.components.sensor] Updating mitemp2_bt sensor took longer than the scheduled update interval 0:00:30
2020-02-27 08:36:17 WARNING (MainThread) [homeassistant.components.sensor] Updating mitemp2_bt sensor took longer than the scheduled update interval 0:00:30
2020-02-27 08:36:18 WARNING (SyncWorker_4) [custom_components.mitemp2_bt.sensor] Polling error Could not read data from Mi Temp sensor A4:C1:38:***
2020-02-27 08:36:28 WARNING (MainThread) [homeassistant.helpers.entity] Update of sensor.sleeping_room_temperature is taking over 10 seconds
2020-02-27 08:36:48 WARNING (MainThread) [homeassistant.components.sensor] Updating mitemp2_bt sensor took longer than the scheduled update interval 0:00:30
2020-02-27 08:37:18 WARNING (SyncWorker_17) [custom_components.mitemp2_bt.sensor] Polling error Could not read data from Mi Temp sensor A4:C1:38:***
2020-02-27 08:37:29 WARNING (MainThread) [homeassistant.helpers.entity] Update of sensor.sleeping_room_humidity is taking over 10 seconds
2020-02-27 08:37:50 WARNING (MainThread) [homeassistant.components.sensor] Updating mitemp2_bt sensor took longer than the scheduled update interval 0:00:30
2020-02-27 08:38:19 WARNING (SyncWorker_6) [custom_components.mitemp2_bt.sensor] Polling error Could not read data from Mi Temp sensor A4:C1:38:***
2020-02-27 08:38:21 WARNING (MainThread) [homeassistant.components.sensor] Updating mitemp2_bt sensor took longer than the scheduled update interval 0:00:30
2020-02-27 08:38:29 WARNING (MainThread) [homeassistant.helpers.entity] Update of sensor.sleeping_room_temperature is taking over 10 seconds
2020-02-27 08:38:52 WARNING (MainThread) [homeassistant.components.sensor] Updating mitemp2_bt sensor took longer than the scheduled update interval 0:00:30
2020-02-27 08:39:19 WARNING (SyncWorker_9) [custom_components.mitemp2_bt.sensor] Polling error Could not read data from Mi Temp sensor A4:C1:38:***
2020-02-27 08:39:33 WARNING (MainThread) [homeassistant.helpers.entity] Update of sensor.sleeping_room_humidity is taking over 10 seconds
2020-02-27 08:39:54 WARNING (MainThread) [homeassistant.components.sensor] Updating mitemp2_bt sensor took longer than the scheduled update interval 0:00:30
`

Did I something misconfigure? - Thanks for your work on the new sensor!!

**Comment by gallynero** [comment]
I'm going to investigate the cause of your error.

**Comment by mbuczkowski-gm** [comment]
I am having the same issue. Hassos running on RPi3B.

**Comment by aluisperezh** [comment]
I had the same error, but it was cuased by the script I was using previously for this running in the background. Stopped the script, rebooted RPi and everything works.

Thanks Gallynero

**Comment by jPPuetz** [comment]
> 
> 
> I had the same error, but it was cuased by the script I was using previously for this running in the background. Stopped the script, rebooted RPi and everything works.
> 
> Thanks Gallynero

I tried rebooting my RPi and removing the devices manually with bluetoothctl. After removing the docker instance did not connect any more to the sensor. After trusting the MAC again, the docker instance connected automatically to the device. Seen on bluetoothctl and on the sensor itself. But it still runs into the timeout. So something during the polling itself or the processing of the the data isn't running correctly. I there is a option to manually query the sensor or get debug information from a special script, I would try that.

**Comment by aluisperezh** [comment]
Yes, I'm seeing a lot of tmeouts too. Values are OK, but tiemouts and this message:

`Updating mitemp2_bt sensor took longer than the scheduled update interval 0:00:30
`

keep happening too often...

**Comment by gallynero** [comment]
I try to discover the issue, while you can try the following command in the system console?

`gatttool --device=A4:C1:38:79:46:64 --addr-type=public --char-desc --adapter=hci0`

You must change the device and adapter that you use in your system.

**Comment by aluisperezh** [comment]
I tried to test it and got this error:

# gatttool --device=A4:C1:38:76:22:EA --addr-type=public --char-desc --adapter=hci0
connect error: Function not implemented (38)

I'm not sure the adapter is correct. How can I know which adapter I'm using? This is a RPi3b+ using the standard Bluetooth adapter in Raspbian

**Comment by aluisperezh** [comment]
Yesterday one of my 2 sensors stopped working in HA: "Disconnected". Today both are disconnected :(

**Comment by jPPuetz** [comment]
gatttool isn't available for Arch arm and deprecated. Is there another option to debug or provide more detailed information for you?

**Comment by cachenow** [comment]
I have the same issue. I added two devices, one works well, another only "battery" works.

**Comment by dawiddydlinski** [comment]
Same as above, only battery works properly
